File: abstractions/java/lib/src/main/java/com/microsoft/kiota/QueryParameter.java
package com.microsoft.kiota;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/*
 * This annotation allows mapping between the query parameter name in the template and the property name in the class.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface QueryParameter {
    /** The name of the parameter in the template */
    public String name();
}

File: abstractions/php/src/QueryParameter.php
<?php
/**
 * Copyright (c) Microsoft Corporation.  All Rights Reserved.
 * Licensed under the MIT License.  See License in the project root
 * for license information.
 */


namespace Microsoft\Kiota\Abstractions;

/**
 * Class QueryParameter
 *
 * Attribute/annotation for query parameter class  properties
 *
 * @Annotation
 * @Target("PROPERTY")
 * @NamedArgumentConstructor
 * @package Microsoft\Kiota\Abstractions
 * @copyright 2022 Microsoft Corporation
 * @license https://opensource.org/licenses/MIT MIT License
 */
class QueryParameter
{
    /**
     * @var string
     */
    public string $name = "";

    public function __construct(string $name)
    {
        $this->name = $name;
    }
}